app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -95,33 +97,43 @@ class WallpaperManager(
         }
     }
 
+    private fun filterExpiredRemoteWallpapers(wallpaper: Wallpaper): Boolean = when (wallpaper) {
+        is Wallpaper.Remote -> {
+            val notExpired = wallpaper.expirationDate?.let { Date().before(it) } ?: true
+            notExpired || wallpaper.name == settings.currentWallpaper
```

Comment:
Drive-by 🚗 : Out of curiosity, if had a remote wallpaper selected at some point, but is no longer selected, would that disappear?

Reply:
Yes, If the wallpaper is expired it will disappear.

Reply:
That's currently the spec. We'll hopefully add some UX around that to make it a little more user-friendly soon. It's easy to imagine improving the persistence layer eventually to allow people to keep them for longer as well

Reply:
Sounds good, thanks that clarifying!